source-sqlserver-batch/discovery.go

```diff
@@ -356,6 +363,21 @@ func discoverColumns(ctx context.Context, db *sql.DB, discoverSchemas []string)
 		}
 		dataType.nullable = isNullable
 
+		// Add length constraints for char/binary columns with defined lengths. A
+		// value of -1 is used for certain large-object types such as XML.
+		if charMaxLength.Valid && charMaxLength.Int64 > 0 {
```

Comment:
A wrinkle: since we're transiting this through JSON Schema, we need to honor how it sees binary data (as base64-encoded strings) and validates maxLength (as the number of UTF-8 codepoints).

That means that `maxLength` of a binary column is actually 4/3 times it's raw binary length.
We should also be setting `contentEncoding: base64` for binary columns.

Within materializations, we would convert going the other way when mapping back to binary.

Reply:
Oh, that's a very good point. I spent all my time thinking about how length constraints actually work for textual data and completely overlooked the base64 encoding on byte data.

Reply:
Okay, getting back to this now. I've updated the discovery logic so the JSON length constraints are `ceil(n/3)*4` characters for the binary(n) column types and verified that we are setting `contentEncoding: base64` correctly.